File: src/main/java/br/com/capela/amicao/modelo/base/Entidade.java
package br.com.capela.amicao.modelo.base;

import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.MappedSuperclass;

@MappedSuperclass
public abstract class Entidade<T> {

	@Id
	@org.hibernate.annotations.GenericGenerator(name = "hibernate-uuid", strategy = "uuid")
	@GeneratedValue(generator = "hibernate-uuid")
	protected Long id;

	public Long getId() {
		return id;
	}

	public abstract Boolean isMesmaEntidade(T outraEntidade);

	@SuppressWarnings("unchecked")
	@Override
	public boolean equals(Object obj) {
		if (this == obj) {
			return true;
		}
		if (obj == null) {
			return false;
		}
		if (!(obj instanceof Entidade<?>)) {
			return false;
		}
		T outro = (T) obj;
		return isMesmaEntidade(outro);
	}

}

File: src/main/scala/br/com/capela/amicao/modelo/base/Entidade.scala
package br.com.capela.amicao.modelo.base;

import javax.persistence._

import scala.reflect._

import _root_.com.google.appengine.api.datastore.Key;

@MappedSuperclass
trait Entidade[T]{

	@Id
	@GeneratedValue{ val strategy=GenerationType.IDENTITY }
   	@BeanProperty var id:Key = _
	
	override def equals(outraEntidade:Any) = outraEntidade match { 
	  case outra: Entidade[T] => outra.getClass == getClass && this.isMesmaEntidade(outra.asInstanceOf[T]);
   	  case _ => false
	}
	
	def isMesmaEntidade(outraEntidade: T):Boolean;
}

